src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,162 @@
+        }
+        if (size == index) {
+            add(value);
+        } else {
+            insertByIndex(value, node(index));
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        return node(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        T oldValue = node(index).item;
+        Node<T> node = node(index);
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> oldElement = node(index);
+        if (size != 1) {
+            Node<T> beforeToElement = oldElement.prev;
+            Node<T> afterToElement = oldElement.next;
+            if (beforeToElement != null) {
+                beforeToElement.next = afterToElement;
+            } else {
+                this.head = afterToElement;
+                afterToElement.prev = null;
+            }
+            if (afterToElement != null) {
+                afterToElement.prev = beforeToElement;
+            } else {
+                this.tail = beforeToElement;
+                beforeToElement.next = null;
+            }
+        }
+        oldElement.next = null;
+        oldElement.prev = null;
+        T element = oldElement.item;
+        oldElement.item = null;
+        size--;
+        return element;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> value = head;
+        for (int i = 0; i < size; i++) {
+            if (object == value.item || object != null && object.equals(value.item)) {
+                remove(i);
+                return true;
+            }
+            value = value.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private boolean addNotFirstValue(T value) {
+        Node<T> tailValue = tail;
+        Node<T> newElement = new Node<>(tailValue, value,null);
+        tail = newElement;
+        if (tailValue == null) {
+            this.tail = newElement;
+        } else {
+            tailValue.next = newElement;
+        }
+        size++;
+        return true;
+    }
+
+    private boolean addFirstValue(T value) {
+        Node<T> headValue = head;
+        Node<T> newElement = new Node<>(null, value,headValue);
+        head = newElement;
+        if (headValue == null) {
+            tail = newElement;
+        } else {
+            headValue.next = newElement;
+        }
+        size++;
+        return true;
+    }
+
+    private Node<T> node(int index) {
```

Comment:
can we make name of method more informative?

Reply:
Yes, we can! I did this!)